sampler.go

```diff
@@ -384,34 +398,57 @@ func (s *httpSamplingManager) GetSamplingStrategy(serviceName string) (*sampling
 	return &out, nil
 }
 
+// SetMaxOperations sets the maximum number of operations the sampler will keep track of.
+func SetMaxOperations(maxOperations int) func(*RemotelyControlledSampler) {
```

Comment:
another problem with these public methods is they allow mutating the sampler after construction